File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        newList = []
        for point in points:
            copy = [point[0], point[1], math.sqrt(point[0]**2 + point[1]**2)]
            newList.append(copy)
        newList = sorted(newList, key= lambda x : x[2])
        ans = []
        for i in range(k):
            point = newList[i]
            ans.append([point[0], point[1]])
        return ans

File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int[]> q = new PriorityQueue<>((a, b) -> {
            int x1 = a[0];
            int y1 = a[1];
            int x2 = b[0];
            int y2 = b[1];
            return (x2 * x2 + y2 * y2) - (x1 * x1 + y1 * y1);
        });
        
        for (int i = 0; i < points.length; i++) {
            q.add(points[i]);
            if (q.size() > k) {
                q.poll();
            }
        }
        
        int[][] ans = new int[k][2];
        for (int i = 0; i < k; i++) {
            ans[i] = q.poll();
        }
        return ans;
    }
}